src/tasks/dump.ts

```diff
@@ -416,6 +415,143 @@ async function transferSameTokenToReceiver(
+  network,
+  hre,
+  api,
+  dryRun,
+  doNotPrompt,
+}: DumpInput): Promise<void> {
+  const { ethers } = hre;
+  if (validity > MAX_ORDER_VALIDITY_SECONDS) {
+    throw new Error("Order validity too large");
+  }
+  const [chainId, allowanceManager] = await Promise.all([
+    ethers.provider.getNetwork().then((n) => n.chainId),
+    (await settlement.allowanceManager()) as string,
+  ]);
+  const domainSeparator = domain(chainId, settlement.address);
+  const receiver: string = inputReceiver ?? signer.address;
+  const hasCustomReceiver = receiver !== signer.address;
+
+  // todo: when sending ETH to a contract will be supported by the
+  // services, remove this check.
+  if (
+    (toTokenAddress === undefined || toTokenAddress === BUY_ETH_ADDRESS) &&
+    hasCustomReceiver
+  ) {
+    throw new Error("Receiver is not supported when buying ETH");
+  }
+
+  const {
+    instructions,
+    toToken,
+    transferToReceiver,
+  } = await getDumpInstructions({
+    dumpedTokens,
+    toTokenAddress,
+    user: signer.address,
+    allowanceManager,
+    maxFeePercent,
+    hasCustomReceiver,
+    hre,
+    network,
+    api,
+  });
+  if (instructions.length === 0) {
+    console.log("No token can be sold");
+    return;
+  }
+
+  displayOperations(instructions, toToken);
+
+  let sumReceived = instructions.reduce(
+    (sum, inst) => sum.add(inst.receivedAmount),
+    constants.Zero,
+  );
+  const needAllowances = instructions
+    .filter(({ needsAllowance }) => needsAllowance)
+    .map(({ token }) => token);
+  if (needAllowances.length !== 0) {
+    console.log(
+      `Before creating the orders, a total of ${needAllowances.length} allowances will be granted to the allowance manager.`,
+    );
+  }
+  const toTokenName = isNativeToken(toToken)
+    ? toToken.symbol
+    : toToken.symbol ?? `units of token ${toToken.address}`;
+  if (transferToReceiver !== undefined) {
+    console.log(
+      `Moreover, ${utils.formatUnits(
+        transferToReceiver.amount,
+        transferToReceiver.token.decimals ?? 0,
+      )} ${toTokenName} will be transfered to the receiver address ${receiver}.`,
+    );
+    sumReceived = sumReceived.add(transferToReceiver.amount);
+  }
+  console.log(
+    `${
+      hasCustomReceiver
+        ? `The receiver address ${receiver}`
+        : `Your address (${signer.address})`
+    } will receive at least ${utils.formatUnits(
+      sumReceived,
+      toToken.decimals ?? 0,
+    )} ${toTokenName} from selling the tokens listed above.`,
+  );
+  if (!dryRun && (doNotPrompt || (await prompt("Submit?")))) {
+    await createAllowances(needAllowances, signer, allowanceManager);
+
+    // If the services don't register the allowance before the order,
+    // then creating a new order with the API returns an error.
+    // Moreover, there is no distinction in the error between a missing
+    // allowance and a failed order creation, which could occur for
+    // valid reasons.
+    await sleepMillis(5000);
```

Comment:
Here as well, it might be better so wait for more confirmations in the createAllowance function rather than relying on a sleep (if not helper method should also be used in other PR).